packages/theme/src/cli/utilities/theme-fs.test.ts

```diff
@@ -471,6 +478,84 @@ describe('theme-fs', () => {
     })
   })
 
+  describe('handleFileDelete', () => {
+    const themeId = '1'
+    const adminSession = {token: 'token', storeFqdn: 'store.myshopify.com'}
+    const root = 'src/cli/utilities/fixtures/theme'
+
+    beforeEach(() => {
+      const mockWatcher = new EventEmitter()
+      vi.spyOn(chokidar, 'watch').mockImplementation((_) => {
+        return mockWatcher as any
+      })
+    })
+
+    test('deletes file from remote theme', async () => {
+      // Given
+      vi.mocked(fetchThemeAsset).mockResolvedValue({
+        key: 'assets/base.css',
+        checksum: '1',
+        value: 'content',
+        attachment: '',
+        stats: {size: 100, mtime: 100},
+      })
+
+      // When
+      const themeFileSystem = mountThemeFileSystem(root)
+      await themeFileSystem.ready()
+
+      const deleteOperationPromise = new Promise<void>((resolve) => {
+        themeFileSystem.addEventListener('unlink', () => {
+          setImmediate(resolve)
+        })
+      })
```

Comment:
FYI there's also [`vi.waitFor` in Vitest](https://vitest.dev/api/vi.html#vi-waitfor) to deal with async code, but I think the promise here is actually better 👍